### PR TITLE
Use a helper to to define the cpp toolchain type

### PR DESCRIPTION
**Pull request by katre**
Part of https://github.com/bazelbuild/bazel/issues/14727

### CODE REVIEW

**Comment by katre** [comment]
@phst PTAL

**Comment by phst** [comment]
I've now dropped compatibility for the Bazel versions that don't support the helper function (that was necessary because µpb now also requires it), could you please rebase and force-push to re-trigger the tests? Thanks!

**Comment by katre** [comment]
I rebased and pushed, but apparently I can't approve the tests.

File: elisp/defs.bzl [hunk 1]
```diff
@@ -540,8 +540,7 @@ The source file is byte-compiled.  At runtime, the compiled version is loaded
 in batch mode unless `interactive` is `True`.""",
     executable = True,
     fragments = ["cpp"],
-    toolchains = [
-        "@bazel_tools//tools/cpp:toolchain_type",
+    toolchains = use_cpp_toolchains() + [
```
Comment (phst) [inline elisp/defs.bzl:543]: Typo :) Same below

Comment (katre) [reply elisp/defs.bzl]: Whoops, fixed.

File: elisp/defs.bzl [hunk 2]
```diff
@@ -280,7 +280,6 @@ def _elisp_test_impl(ctx):
     ]
 
 elisp_toolchain = rule(
-    implementation = _elisp_toolchain_impl,
```
Comment (phst) [inline elisp/defs.bzl:283]: Please revert these formatting changes.

Comment (katre) [reply elisp/defs.bzl]: Done. This was applied by vim running buildifier on all bzl files automatically, sorry about that.

**Review by phst — APPROVED** [review]

**Pull request merged**